x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/actions.tsx

```diff
@@ -58,20 +58,29 @@ export const duplicateRulesAction = async (
     } else {
       displaySuccessToast(i18n.SUCCESSFULLY_DUPLICATED_RULES(ruleIds.length), dispatchToaster);
     }
-    dispatch({ type: 'loadingRuleIds', ids: [], actionType: null });
-
     return createdRules;
   } catch (error) {
-    dispatch({ type: 'loadingRuleIds', ids: [], actionType: null });
     errorToToaster({ title: i18n.DUPLICATE_RULE_ERROR, error, dispatchToaster });
+  } finally {
+    dispatch({ type: 'loadingRuleIds', ids: [], actionType: null });
   }
 };
 
-export const exportRulesAction = (
+export const exportRulesAction = async (
   exportRuleId: string[],
-  dispatch: React.Dispatch<RulesTableAction>
+  dispatch: React.Dispatch<RulesTableAction>,
+  dispatchToaster: Dispatch<ActionToaster>
 ) => {
-  dispatch({ type: 'exportRuleIds', ids: exportRuleId });
+  try {
+    dispatch({ type: 'loadingRuleIds', ids: exportRuleId, actionType: 'export' });
+    const blob = await exportRules({ ids: exportRuleId });
+    downloadBlob(blob, `${i18n.EXPORT_FILENAME}.ndjson`);
+    displaySuccessToast(i18n.SUCCESSFULLY_EXPORTED_RULES(exportRuleId.length), dispatchToaster);
```

Comment:
FYI, `displaySuccessToast` and `displayErrorToast` are deprecated. There are hook versions of the notifications throughout the code base.

You don't have to change it here, but when possible we should try to reduce/remove/replace these usages.

Reply:
Yeah, I don't use these methods when I write new code and usually replace them when it doesn't take too long.

Reply:
Thanks, appreciate it!